File: app/src/main/java/com/tree/myview/activity/LoopActivity.java
package com.tree.myview.activity;

import android.util.Log;
import android.view.View;
import android.widget.ImageView;

import com.tree.myview.R;

import androidx.constraintlayout.helper.widget.Carousel;

/*
 *  @项目名：  RulerDemo
 *  @包名：    com.tree.myview.activity
 *  @文件名:   LoopActivity
 *  @创建者:   rookietree
 *  @创建时间:  2023/3/10 15:46
 *  @描述：    TODO
 */
public class LoopActivity extends BaseActivity {

    private final int[] imgs = {R.mipmap.meinv, R.mipmap.meinv4, R.mipmap.meinv5, R.mipmap.meinv6,
            R.mipmap.meinv10, R.mipmap.meinv11, R.mipmap.meinv12};

    @Override
    public int getLayoutId() {
        return R.layout.activity_loop2;
    }

    @Override
    public void init() {
        Carousel carousel = findViewById(R.id.carousel);
        carousel.setAdapter(new Carousel.Adapter() {
            @Override
            public int count() {
                return imgs.length;
            }

            @Override
            public void populate(View view, int index) {
                if (view instanceof ImageView) {
                    ((ImageView) view).setImageResource(imgs[index]);
                }
                Log.d("loopww","populate index:"+index);
            }

            @Override
            public void onNewItem(int index) {
                Log.d("loopww","onNewItem index:"+index);
            }
        });
    }
}

File: app/src/main/java/com/tree/myview/activity/LoopActivity.kt
package com.tree.myview.activity

import android.util.Log
import android.view.View
import android.widget.ImageView
import androidx.constraintlayout.helper.widget.Carousel
import coil.load
import coil.size.Scale
import com.tree.myview.R

/*
 *  @项目名：  RulerDemo
 *  @包名：    com.tree.myview.activity
 *  @文件名:   LoopActivity
 *  @创建者:   rookietree
 *  @创建时间:  2023/3/10 15:46
 *  @描述：
 */
class LoopActivity : BaseActivity() {
    private val imgs = intArrayOf(
        R.drawable.look1,
        R.drawable.look2,
        R.drawable.look3,
        R.drawable.look4,
        R.drawable.look5
    )

    override fun getLayoutId(): Int {
        return R.layout.activity_loop
    }

    override fun init() {
        val carousel = findViewById<Carousel>(R.id.carousel)
        carousel.setAdapter(object : Carousel.Adapter {
            override fun count(): Int {
                return imgs.size
            }

            override fun populate(view: View, index: Int) {
                if (view is ImageView) {
                    view.setImageResource(imgs[index])
                }
            }

            override fun onNewItem(index: Int) {
            }
        })
    }
}